stores/redis/redis.go

```diff
@@ -0,0 +1,344 @@
+package redis
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis"
+	"github.com/mxschmitt/golang-url-shortener/stores/shared"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	entryPathPrefix     = string("entry:")       // prefix for path-to-url mappings
+	entryUserPrefix     = string("user:")        // prefix for path-to-user mappings
+	userToEntriesPrefix = string("userEntries:") // prefix for user-to-[]entries mappings (redis SET)
+	entryVisitsPrefix   = string("entryVisits:") // prefix for entry-to-[]visit mappings (redis LIST)
+)
+
+// RedisStore implements the stores.Storage interface
+type RedisStore struct {
+	c *redis.Client
+}
+
+func New(hostaddr, password string) (*RedisStore, error) {
+	c := redis.NewClient(&redis.Options{
+		Addr:     hostaddr,
+		Password: password,
+		DB:       0,
+	})
+	// if we can't talk to redis, fail fast
+	_, err := c.Ping().Result()
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not connect to redis db0")
+	}
+	ret := &RedisStore{c: c}
+	return ret, nil
+}
+
+func (r *RedisStore) keyExists(key string) (exists bool, err error) {
+	var result *redis.IntCmd
+	logrus.Debugf("Checking for existence of key: %s", key)
+	result = r.c.Exists(key)
```

Comment:
I would simplify the Variable declarations to `:=` and remove the declaration 2 lines above.

Reply:
done